pubsub.go

```diff
@@ -532,6 +542,23 @@ func (c *PubSub) ChannelWithSubscriptions(opts ...ChannelOption) <-chan interfac
 	return c.allCh.allCh
 }
 
+func (c *PubSub) processPendingPushNotificationWithReader(ctx context.Context, cn *pool.Conn, rd *proto.Reader) error {
+	if c.pushProcessor == nil {
+		return nil
+	}
+
+	// Create handler context with client, connection pool, and connection information
+	handlerCtx := c.pushNotificationHandlerContext(cn)
+	return c.pushProcessor.ProcessPendingNotifications(ctx, handlerCtx, rd)
+}
+
+func (c *PubSub) pushNotificationHandlerContext(cn *pool.Conn) push.NotificationHandlerContext {
+	// PubSub doesn't have a client or connection pool, so we pass nil for those
+	// PubSub connections are blocking
+	return push.HandlerContext{}
+	return push.NewNotificationHandlerContext(nil, nil, c, cn, true)
```

Comment:
@ndyakov refactor. Remove the interface and use a concrete type for the HandlerContext. It will improve readability.